threedi_api_qgis_client/widgets/wizard.py

```diff
@@ -469,6 +580,233 @@ def plot_overview_precipitation(self):
+        self._init_widget()
+        self._connect_signals()
+
+    def _init_widget(self):
+        self.overrule_widget.setVisible(False)
+        self.cb_type.addItem("1D")
+        self.cb_type.addItem("2D")
+        self.dd_units.addItem("s")
+
+    def _connect_signals(self):
+        self.checkBox_1.stateChanged.connect(self.overrule_value_canged)
+        self.pb_upload_1.clicked.connect(self.open_upload_dialog)
+        self.pb_save.clicked.connect(self.save_laterals)
+        self.cb_type.currentIndexChanged.connect(self.selectionchange)
+        self.cb_laterals.activated.connect(self.laterals_change)
+
+    def laterals_change(self):
+        lat_id = self.cb_laterals.currentText()
+        lat =self.laterals_timeseries.get(lat_id)
+        self.il_location.setText(" ,".join(str(a) for a in lat.values[0]))
+        self.il_discharge.setText(" ,".join(str(a) for a in lat.values[1]))
+        self.sb_offset.setValue(lat.offset)
+
+    def save_laterals(self):
+        lat = self.laterals_timeseries.get(self.cb_laterals.currentText())
+        lat.values[0] = [float(f) for f in self.il_location.text().split(",")]
+        lat.values[1] = [float(f) for f in self.il_discharge.text().split(",")]
+        lat.offset(self.sb_offset.value())
+
+    def selectionchange(self, index):
+        if index == 0:
+            self.laterals_layout.setText("Upload laterals for 1D:")
+        if index == 1:
+            self.laterals_layout.setText("Upload laterals for 2D:")
+
+    def open_upload_dialog(self, param):
+        last_folder = QSettings().value("threedi/last_precipitation_folder", os.path.expanduser("~"), type=str)
+        file_filter = "CSV (*.csv );;All Files (*)"
+        filename, __ = QFileDialog.getOpenFileName(self, "Precipitation Time Series", last_folder, file_filter)
+        if len(filename) == 0:
+            return
+        QSettings().setValue("threedi/last_precipitation_folder", os.path.dirname(filename))
+        with open(filename) as lateral_file:
+            # todo parse csv
+            laterals_reader = csv.reader(lateral_file)
+            if self.cb_type.currentText() == "1D":
+                values = {}
+                for row_id, id,  connection_node_id, timeseries in laterals_reader:
+                    if timeseries == "timeseries":
+                        continue
+                    try:
+                        timeseries_parsed = [float(f) for f in timeseries.split("\n")[0].split(",")]
+                        discharge = [float(f) for f in timeseries.split("\n")[1].split(",")]
+                        lateral = TimeseriesLateral(values=[timeseries_parsed, discharge], units="m3/s", point=None, connection_node=int(connection_node_id), id=int(id), offset=0)
+                        values[id] = lateral
+                    except ValueError:
+                        continue
+            if self.cb_type.currentText() == "2D":
+                for x, y,  type, id, timeseries in laterals_reader:
```

Comment:
Please avoid shadowing  built-in 'type',  'id'. There is also extra whitespace after 'y'.